### PR TITLE
StatPanels: Add new calculation option for percentage difference 

### PR DESCRIPTION
**Pull request by jedstar**
**What this PR does / why we need it**:
Add a calculation to the STAT panel for 'Percentage Difference'

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This is my first time contributing using Git, so I apologize if I have missed anything in the procedures.
I've been using this calculation in a custom docker build of grafana 6.6 for a while as I really needed a stat of percentage difference that dynamically updated with the time range.

Example1:
first value = 100
last value = 110
Difference Percentage = 10%

Example2:
first value = 1000
last value = 1010
Difference Percentage = 1%

### CODE REVIEW

File: docs/sources/panels/calculations-list.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Among other places, these calculations are used in the Transform tab and the bar
 | Count | number of values in a field|
 | Delta | cumulative change in value |
 | Difference | difference between first and last value of a field|
+| Difference Percent | percentage change between first and last value of a field |
```
Comment (oddlittlebird) [inline docs/sources/panels/calculations-list.md:24]:
```suggestion
| Difference percent | percentage change between first and last value of a field |
```

Comment (oddlittlebird) [reply docs/sources/panels/calculations-list.md:24]: Please update the UI as well.

File: public/app/plugins/panel/singlestat/module.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ class SingleStatCtrl extends MetricsPanelCtrl {
     { value: 'first', text: 'First' },
     { value: 'delta', text: 'Delta' },
     { value: 'diff', text: 'Difference' },
+    { value: 'diffperc', text: 'Difference Percentage' },
```
Comment (oddlittlebird) [inline public/app/plugins/panel/singlestat/module.ts:71]:
```suggestion
    { value: 'diffperc', text: 'Difference percentage' },
```

**Review by torkelo — APPROVED** [review]
Looks good but the name could be better maybe?

File: packages/grafana-data/src/transformations/fieldReducer.ts [hunk 3]
```diff
@@ -223,6 +224,12 @@ export const fieldReducers = new Registry<FieldReducerInfo>(() => [
     standard: false,
     reduce: calculateDistinctCount,
   },
+  {
+    id: ReducerID.diffperc,
+    name: 'Difference Percentage',
```
[the remark this answers is not in the document]
Comment (oddlittlebird) [reply packages/grafana-data/src/transformations/fieldReducer.ts:229]: Agreed. Sentence case would keep it consistent with design and documentation guidelines.

[the remark this answers is not in the document]
Comment (dprokop) [reply packages/grafana-data/src/transformations/fieldReducer.ts:229]: I also agree with this, and would maybe alternatively go with naming it `Difference(%)`?

File: packages/grafana-data/src/transformations/fieldReducer.ts [hunk 4]
```diff
@@ -351,6 +359,7 @@ export function doStandardCalcs(field: Field, ignoreNulls: boolean, nullAsZero:
 
   if (isNumber(calcs.firstNotNull) && isNumber(calcs.lastNotNull)) {
     calcs.diff = calcs.lastNotNull - calcs.firstNotNull;
+    calcs.diffperc = calcs.diff / calcs.firstNotNull;
```
Comment (dprokop) [inline packages/grafana-data/src/transformations/fieldReducer.ts:362]: Maybe a test for this calculation would be a good idea, just to make sure no one breaks this in the future. You calc add it in the fieldReducer.test.ts

**Review by dprokop — APPROVED** [review]
Left few nits to consider, other than that thanks a lot for the contribution!

**Comment by dprokop** [comment]
@jedstar any updates on this one?

**Comment by aknuds1** [comment]
For the CI errors, try merging with latest master.

**Comment by aknuds1** [comment]
@dprokop I think this could use another review?

**Pull request merged**